js-mentoring-program-master/tasks/cycles/cycles.js

```diff
@@ -23,13 +31,40 @@ function calculateDiscount(redemption) {
  */
 {
   const i = 10; //10! = 3628800
+  let fact = 1;
+
+// for
+    for (let i = 1; i <= 10; i++) {
+        fact *= i;
+    }
+
+// while
+    let index = 1;
+    fact = 1;
+    while (index <= 10) {
+        fact *= index;
+        index++;
+    }
+
+    index = 1;
+    fact = 1;
+// do .. while
+    do {
+        fact *= index;
+        index++;
+    } while (index <= 10);
 }
 
 /**
  * return concatenated string from an array of substring
  */
 {
   const substr = ["I", " love", " JS"];
+  let str = '';
+
+  for (let i = 0; i < substr.length; i++) {
+    str+=substr[i];
+}
```

Comment:
Look up an appropriate mathod in Array.prototype to transform all the elements in one string.